src/Sulu/Component/HttpCache/Handler/PublicHandler.php

```diff
@@ -59,6 +59,11 @@ public function updateResponse(Response $response, StructureInterface $structure
             return;
         }
 
+        // when structure cache-lifetime disabled - return
+        if ((int) $structure->getCacheLifeTime() === 0) {
+            return;
+        }
```

Comment:
@chirimoya Is this a real fix for you? For me it looks more like the values of `$maxAge` and `$sharedMaxAge` are the problem. I guess if we have a cache lifetime of 60 seconds, this would probably still lead to a strange behavior... And this solution only allows to set the cache lifetime in the controller, if the template has a cache lifetime of 0, I think we should find a better solution than that.